UI: Ordering Table enable no entries cell and disable ordering on no data

fhelfer opened the pull request:
DataTable already has a full width cell if there are no entries, but OrderingTable currently only displays the header row. Also deactivates Ordering Multi-Action Button and JavaScript if there are no entries.

Added example and changed test accordingly
<img width="323" height="268" alt="image" src="https://github.com/user-attachments/assets/e49d4a74-f969-4318-9449-0af7c5e5698f" />


Let me know what you think of this proposal

Best @fhelfer

fhelfer committed:
UI: Ordering Table enable no entries cell, disable ordering

fhelfer commented:
https://mantis.ilias.de/view.php?id=45729
https://mantis.ilias.de/view.php?id=45730

(edit: also a duplicate in https://mantis.ilias.de/view.php?id=45309)

fhelfer committed:
UI: Ordering Table missing COL_SPAN

fhelfer committed:
UI: Ordering Table bugfix action row has not always full width

iszmais commented:
@fhelfer Thank you for your Contribution!

I assume this would need some approval from the JF even if your changes are very intuitive, since I'm not entirly sure if this might correlate with other concepts.

If this is approved and no one other shows interest in herding it i will gladly do so.

Greetings,
@iszmais

thibsy commented:
Hi @fhelfer and @iszmais,

This should not have been on the shepherd list. The (remaining) UI-Coordinators will have a look at this. I can't really say how fast yet though.

I also removed the JF label, we will do so after our review.

Kind regards,
@thibsy

dsstrassner commented:
I think @oliversamoila is also added to the UI Coordinators last week from the PM and TB. I am too lazy to search for the corresponding Pull Request. 

@matthiaskunkel & @ILIAS-eLearning/technical-board maybe it is a good idea to set a rule for the commit message 'title' other than "update maintenance.md" for those authority changes 😬

thibsy commented:
@dsstrassner I raise you one: instead of one rule for commit messages changing authorities, it would be even better to have some commit-message schema in general =). Like we (sr solution) use something like `[FIX|FEATURE] #<issue-nr> <component-name>: <message>`.

oliversamoila reviewed (changes requested):
Hello @fhelfer,
Thank you very much for your contribution.

When I look at the 'Test' as an example, I still get the Bulk Actions offered regardless of the empty list of all questions. Could you take another look at this? 
(This only affects the Ordering Table, not the UI Data Table.)

Thank you very much.
Best regards, 
@oliversamoila 

> <img width="892" height="520" alt="Bildschirmfoto 2025-12-29 um 08 43 22" src="https://github.com/user-attachments/assets/f2dfa3f2-5ea9-4937-aa3b-5acd68d65b81" />

fhelfer committed:
[Fix] OrderingTable: disable multiaction row if no data is available

fhelfer commented:
Thank you for testing this @oliversamoila,
I changed it now so that the entire row is not rendered if there is no data

I also rebased the PR

Best  @fhelfer

fhelfer committed:
[Fix] OrderingTable: UnitTest testOrderingTableRenderTableHeaderWithoutActionsWithoutData

oliversamoila reviewed (approved):
Hello together, 
from a UX and UI perspective, the result looks good. Thank you very much for your contribution.
I'm forwarding this to @thibsy for code review. He may contact you with change requests or incorporate the changes directly (That includes cherry picking too.)

Kind regards,
@oliversamoila (as UI coordinator)
